pkg/solana/logpoller/log_poller.go

```diff
@@ -201,6 +213,44 @@ func (lp *Service) UnregisterFilter(ctx context.Context, name string) error {
 	return lp.filters.UnregisterFilter(ctx, name)
 }
 
+// Replay submits a new replay request. If there was already a new replay request
+// submitted since the last replay completed, it will be updated to the earlier of the
+// two requested fromBlock's. The expectation is that, on the next timer tick of the
+// LogPoller run loop it will backfill all filters starting from fromBlock. If there
+// are new filters in the backfill queue, with an earlier StartingBlock, then they
+// will get backfilled from there instead.
+func (lp *Service) Replay(ctx context.Context, fromBlock int64) error {
+	ctx, cancel := lp.eng.Ctx(ctx)
+	defer cancel()
+
+	lp.replay.mut.Lock()
+	defer lp.replay.mut.Unlock()
+
+	if lp.replay.requestBlock != NoNewReplayRequests && lp.replay.requestBlock <= fromBlock {
+		// Already requested, no further action required
```

Comment:
Should we log warning that replay request was ignored?

Reply:
Sure, I can add that